packages/filebrowser-extension/src/index.ts

```diff
@@ -479,6 +482,66 @@ const downloadPlugin: JupyterFrontEndPlugin<void> = {
   }
 };
 
+/**
+ * A plugin to add the "Open in Terminal" command to the file browser context menu.
+ */
+const openInTerminalPlugin: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlab/filebrowser-extension:open-in-terminal',
+  description:
+    'Adds a context menu item to open a terminal in the file browser directory.',
+  requires: [IFileBrowserFactory, ITranslator],
+  autoStart: true,
+  activate: (
+    app: JupyterFrontEnd,
+    factory: IFileBrowserFactory,
+    translator: ITranslator
+  ): void => {
+    const { commands } = app;
+    const { tracker } = factory;
+    const trans = translator.load('jupyterlab');
+
+    commands.addCommand(CommandIDs.openInTerminal, {
+      label: trans.__('Open in Terminal'),
+      icon: terminalIcon.bindprops({ stylesheet: 'menuItem' }),
+      execute: async () => {
+        const widget = tracker.currentWidget;
+        if (!widget) {
+          return;
+        }
+
+        const item = widget.selectedItems().next().value;
+        if (!item) {
+          return;
+        }
+
+        let path = '';
+        if (item.type === 'directory') {
+          path = item.path;
+        } else {
+          path = PathExt.dirname(item.path);
+        }
+
+        try {
+          const terminal = await commands.execute('terminal:create-new', {
+            cwd: path
+          });
+          if (terminal) {
+            app.shell.activateById(terminal.id);
+          }
+        } catch (e) {
+          console.error(`Failed to open terminal in '${path}'`, e);
+        }
+      }
+    });
+
+    app.contextMenu.addItem({
+      command: CommandIDs.openInTerminal,
+      selector: '.jp-DirListing-item',
+      rank: 2 // positioning it near the top of the menu(after menu)"
+    });
```

Comment:
We prefer the context menu definitions in JSON settings, this allows users to disable them in settings  https://jupyterlab.readthedocs.io/en/stable/extension/extension_points.html#context-menu

I think I would put it lower down personally but that's not a very strong opinion.